osquery/dispatcher/tests/scheduled_tasks_queue_tests.cpp

```diff
@@ -0,0 +1,136 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <chrono>
+#include <iostream>
+#include <thread>
+
+#include <gtest/gtest.h>
+
+#include <osquery/config.h>
+#include <osquery/logger.h>
+#include <osquery/system.h>
+
+#include "osquery/dispatcher/scheduled_tasks_queue.h"
+
+namespace osquery {
+
+DECLARE_bool(disable_logging);
+
+class ScheduledTasksQueueTests : public testing::Test {
+  void SetUp() override {
+    logging_ = FLAGS_disable_logging;
+    FLAGS_disable_logging = true;
+  }
+
+  void TearDown() override {
+    FLAGS_disable_logging = logging_;
+  }
+
+ private:
+  bool logging_{false};
+};
+
+TEST_F(ScheduledTasksQueueTests, run_for_the_one_time) {
+  auto tasks = ScheduledTaskQueue{};
+  auto counter = int{0};
```

Comment:
I'm actually not sure here, but is there any benefit in using auto if you're going to define the type with the right hand side anyway?

Reply:
This way is more type strictly. Look at this.
```
unsigned num = -9;
```
This code will be compiled without any errors and warnings. Which is wrong I guess.
With auto and brace initialization it won't be compiled with reasonable error.
```
auto num = unsigned{-9};
```

It looks pretty stupid and you may think it is easy to avoid it. Furthermore it is `0`, it could be noting wrong with it.
But on practice vast majority of bugs is stupid. So brace initialization with auto is fair way to avoid some of them.
Does it make sense?